openmc/mgxs/library.py

```diff
@@ -1005,32 +1018,79 @@ def create_mg_library(self, xs_type='macro', xsdata_names=None,
         # Initialize file
         mgxs_file = openmc.MGXSLibrary(self.energy_groups)
 
-        # Create the xsdata object and add it to the mgxs_file
-        for i, domain in enumerate(self.domains):
-            if self.by_nuclide:
-                nuclides = list(domain.get_all_nuclides().keys())
+        if xs_ids is not None:
+            if isinstance(xs_ids, basestring):
+                # If we only have a string lets convert it now to a list
+                # of strings.
+                xs_id_input = xs_ids
+                xs_ids = []
+                for domain in self.domains:
+                    if self.domain_type is 'mesh':
+                        for subdomain in domain.cell_generator():
```

Comment:
This seems like a very slow way to populate the `xs_ids` list - can you simply determine the number of mesh cells in `domain` and then do:

```python
xs_ids = [xs_id_input] * num_mesh_cells
```